core/src/main/java/org/jdbi/v3/core/mapper/reflect/ConstructorMapper.java

```diff
@@ -267,6 +275,9 @@ public T map(ResultSet rs, StatementContext ctx) throws SQLException {
 
             for (int i = 0; i < count; i++) {
                 params[i] = mappers[i].map(r, c);
+                if (propagateNulls[i] && (params[i] == null || isPrimitive[i] && r.wasNull())) {
```

Comment:
I refactored all this code to use a struct instead of endlessly growing parallel arrays -- and even uncovered a bug in the process :)

made sorting these to the front much easier, so I added that as well.